### PR TITLE
#252: Remove auto-refresh tags from page snapshots

### PR DESCRIPTION
**Pull request by jowerner**

### CODE REVIEW

File: src/main/java/com/xceptance/xlt/engine/resultbrowser/PageTransformer.java [hunk 1]
```diff
@@ -159,9 +159,13 @@ Document transform(final UrlMapping mapping)
                 for (int j = 0; j < attributes.getLength(); j++)
                 {
                     final Attr attribute = (Attr) attributes.item(j);
-                    if (attribute.getName().equalsIgnoreCase("http-equiv") && attribute.getValue().equalsIgnoreCase("content-type"))
+                    if (attribute.getName().equalsIgnoreCase("http-equiv"))
                     {
-                        metaTag.getParentNode().removeChild(metaTag);
+                        final String attributeValue = attribute.getValue();
+                        if (attributeValue.equalsIgnoreCase("content-type") || attributeValue.equalsIgnoreCase("refresh"))
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/engine/resultbrowser/PageTransformer.java:165]: We should adjust comment at line 152.

File: src/test/java/com/xceptance/xlt/engine/resultbrowser/PageTransformerTest.java [hunk 2]
```diff
@@ -278,6 +279,31 @@ public void testTransformLWPage_BaseUrlHandling() throws Throwable
...
+            final PageDOMClone clone = DomUtils.clonePage(page);
+            final PageTransformer instance = new PageTransformer(clone, true);
+            final UrlMapping urlMapping = new DumpMgr().getUrlMapping();
+
+            final Document doc = instance.transform(urlMapping);
+
+            // check that the refresh http-equiv meta tag has been removed, but not the other two http-equiv metas
+            final NodeList metas = doc.getElementsByTagName("meta");
+            Assert.assertEquals(2, metas.getLength());
+            Assert.assertEquals("content-type", ((Element) metas.item(0)).getAttribute("http-equiv"));
```
Comment (h-arlt) [inline src/test/java/com/xceptance/xlt/engine/resultbrowser/PageTransformerTest.java:302]: If I'm not mistaken, `PageTransformer` should remove all meta tags whose `http-equiv` attribute has value `refresh` or `content-type`. Thus, only one meta tag should remain after transformation.

Comment (h-arlt) [reply src/test/java/com/xceptance/xlt/engine/resultbrowser/PageTransformerTest.java:302]: Should have read the entire method body; there is a meta tag added later on.

File: src/main/java/com/xceptance/xlt/engine/resultbrowser/PageTransformer.java (hunk 1 above)
Comment (jowerner) [reply src/main/java/com/xceptance/xlt/engine/resultbrowser/PageTransformer.java:165]: Done.

**Review by h-arlt — APPROVED** [review]

**Pull request merged**